src/bin/pgcopydb/ld_wal2json.c

```diff
@@ -274,6 +268,42 @@ parseWal2jsonMessage(StreamContext *privateContext,
 	return true;
 }
 
+/*
+ * SetSchemaAndTableName parses the table's nspname and relname from the JSON
+ * object and escapes it appropriately to be put as it is in SQL statements
+ */
+static bool
+SetSchemaAndTableName(JSON_Object *jsobj,
+					  LogicalMessageRelation *table,
+					  PGSQL *pgsql)
+{
+	char *schema = NULL;
+	char *relname = NULL;
+
+	schema = (char *) json_object_dotget_string(jsobj, "message.schema");
+	relname = (char *) json_object_dotget_string(jsobj, "message.table");
+
+
+	if (schema == NULL || relname == NULL)
+	{
+		return false;
+	}
+
+	table->nspname = escape_identifier(pgsql, schema);
+	if (table->nspname == NULL)
+	{
+		return false;
+	}
+
+	table->relname = escape_identifier(pgsql, relname);
+	if (table->relname == NULL)
+	{
+		free(table->nspname);
```

Comment:
We need to use `PQfreemem` here certainly, right?

Reply:
Yes documentation suggests to use `PQfreemem` but it felt out of place (I mean most libpq stuff are used within pgsql.c) so I simply used `free` after looking at it's implementation (below). But I agree and understand we should the suggested API

```c
/*
 *		PQfreemem - safely frees memory allocated
 *
 * Needed mostly by Win32, unless multithreaded DLL (/MD in VC6)
 * Used for freeing memory from PQescapeBytea()/PQunescapeBytea()
 */
void
PQfreemem(void *ptr)
{
	free(ptr);
}
```